bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/IndexInMemPageMgr.java

```diff
@@ -51,14 +60,20 @@
 
     private static class InMemPageCollection implements LEPStateChangeCallback {
 
-        ConcurrentMap<Long, ConcurrentMap<Long,LedgerEntryPage>> pages;
+        final ConcurrentMap<Long, ConcurrentMap<Long,LedgerEntryPage>> pages;
+        final Map<EntryKey, LedgerEntryPage> lruCleanPageMap;
+        final ConcurrentLinkedQueue<LedgerEntryPage> listOfFreePages;
```

Comment:
most of the lru logic is already in place. using a list is just a simple and straightforward step. I am not sure if apache commons pool will buy us more here (due to my limited knowledges on apache commons pool).

also keep in mind, this is the code from two years ago. I was just porting from twitter branch. if using apache commons pool can gain more performance, feel free to improve it.